reckless_github/src/repository.rs

```diff
@@ -57,68 +51,56 @@ impl Github {
-                                PluginLang::Python,
-                            ));
-                        }
-                        "go.mod" => {
-                            self.plugins.push(Plugin::new(
-                                &plugin_name,
-                                &path_to_plugin,
-                                PluginLang::Go,
-                            ));
-                        }
-                        "cargo.toml" => {
-                            self.plugins.push(Plugin::new(
-                                &plugin_name,
-                                &path_to_plugin,
-                                PluginLang::Rust,
-                            ));
-                        }
-                        "pubspec.yaml" => {
-                            self.plugins.push(Plugin::new(
-                                &plugin_name,
-                                &path_to_plugin,
-                                PluginLang::Dart,
-                            ));
-                        }
-                        "package.json" => {
-                            self.plugins.push(Plugin::new(
-                                &plugin_name,
-                                &path_to_plugin,
-                                PluginLang::JavaScript,
-                            ));
-                        }
-                        "tsconfig.json" => {
-                            // FIXME: avoid to unwrap here
-                            self.plugins.push(Plugin::new(
-                                plugin_name.as_str(),
-                                path_to_plugin.as_str(),
-                                PluginLang::TypeScript,
-                            ));
-                        }
-                        "reckless.yml" | "reckless.yaml" => {
-                            let conf_path = format!("{}/{}", path_to_plugin, file_name);
-                            let mut conf_str = String::new();
-                            File::open(conf_path.as_str())
-                                .await?
-                                .read_to_string(&mut conf_str)
-                                .await?;
-                            let _: Conf = serde_yaml::from_str(&conf_str).unwrap();
-                            // FIXME: store the conf inside the plugin
-                        }
-                        _ => continue,
+        // We are filtering to not iterate through files or any directories such as .git, .ci, .github
+        for plugin_dir in WalkDir::new(repo_path)
+            .max_depth(1)
+            .into_iter()
+            .filter_entry(|dir_entry| {
+                dir_entry
+                    .file_name()
+                    .to_str()
+                    .map(|s| {
+                        dir_entry.depth() == 0 || !s.starts_with(".") && dir_entry.path().is_dir()
+                    })
+                    .unwrap()
+            })
+        {
```

Comment:
This looks really unreadable

This is an example from the docs that it is much nicer https://docs.rs/walkdir/latest/walkdir/#example-skip-hidden-files-and-directories-on-unix

Reply:
This one is very similar to how we do, and I am presenting a case against using a separate filter function. I have cleaned this up a bit, but if you really want to add a filter function for this, let me know.

Reply:
No it is not, you are doing all inside the for and the code example is splitted in function and variable declared outside the for

Reply:
I suggested a new solution to make the for much readble